flake.nix

```
@@ -1,103 +1,116 @@
 
+          # Typst files to include in the derivation.
+          # Here we include Rust files, assets and tests.
           src = sourceByRegex ./. [
             "(assets|crates|tests)(/.*)?"
             ''Cargo\.(toml|lock)''
             ''build\.rs''
           ];
 
-          cargoLock = {
-            lockFile = ./Cargo.lock;
-            allowBuiltinFetchGit = true;
+          # Typst derivation's args, used within crane's derivation generation
+          # functions.
+          commonCraneArgs = {
+            inherit src pname version;
+
+            buildInputs = optionals pkgs.stdenv.isDarwin [
+              pkgs.darwin.apple_sdk.frameworks.CoreServices
+            ];
+
+            nativeBuildInputs = [ pkgs.installShellFiles ];
           };
 
-          nativeBuildInputs = [
-            pkgs.installShellFiles
-          ];
+          # Derivation with just the dependencies, so we don't have to keep
+          # re-building them.
+          cargoArtifacts = craneLib.buildDepsOnly commonCraneArgs;
 
-          buildInputs = optionals pkgs.stdenv.isDarwin [
-            pkgs.darwin.apple_sdk.frameworks.CoreServices
-          ];
+          typst = craneLib.buildPackage (commonCraneArgs // {
+            inherit cargoArtifacts;
 
-          postInstall = ''
-            installManPage crates/typst-cli/artifacts/*.1
-            installShellCompletion \
-              crates/typst-cli/artifacts/typst.{bash,fish} \
-              --zsh crates/typst-cli/artifacts/_typst
-          '';
+            postInstall = ''
+              installManPage crates/typst-cli/artifacts/*.1
+              installShellCompletion \
+                crates/typst-cli/artifacts/typst.{bash,fish} \
+                --zsh crates/typst-cli/artifacts/_typst
+            '';
 
-          GEN_ARTIFACTS = "artifacts";
-        };
+            GEN_ARTIFACTS = "artifacts";
+          });
+        in
+        typst;
     in
-    {
-      devShells = eachSystem (pkgs: {
-        default = pkgs.mkShell {
-          packages =
-            let
-              fenix' = fenix.packages.${pkgs.stdenv.hostPlatform.system};
-            in
-            [
-              (fenix'.default.withComponents [
-                "cargo"
-                "clippy"
-                "rustc"
-                "rustfmt"
-              ])
-              fenix'.rust-analyzer
-            ];
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      systems = [
+        "aarch64-darwin"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "x86_64-linux"
+      ];
 
-          buildInputs = optionals pkgs.stdenv.isDarwin [
-            pkgs.darwin.apple_sdk.frameworks.CoreServices
-            pkgs.libiconv
-          ];
+      flake = {
+        overlays.default = _: prev: {
+          typst-dev = packageFor prev;
         };
-      });
+      };
 
-      formatter = eachSystem (pkgs: pkgs.nixpkgs-fmt);
+      perSystem = { pkgs, ... }:
+        let
+		  inherit (pkgs) lib;
```

quick indentation fix :)
```suggestion
          inherit (pkgs) lib;
```